internal/component/otelcol/receiver/receiver.go

```diff
@@ -152,19 +168,25 @@ func (r *Receiver) Update(args component.Arguments) error {
 		},
 	}
 
-	receiverConfig, err := rargs.Convert()
+	receiverConfig, err := r.args.Convert()
 	if err != nil {
 		return err
 	}
 
-	next := rargs.NextConsumers()
+	next := r.args.NextConsumers()
 
 	// Create instances of the receiver from our factory for each of our
 	// supported telemetry signals.
 	var components []otelcomponent.Component
 
+	liveDebuggingActive := r.debugDataPublisher.IsActive(livedebugging.ComponentID(r.opts.ID))
+
 	if len(next.Traces) > 0 {
-		nextTraces := fanoutconsumer.Traces(next.Traces)
+		traces := next.Traces
+		if liveDebuggingActive {
+			traces = append(traces, r.liveDebuggingConsumer)
+		}
+		nextTraces := fanoutconsumer.Traces(traces)
```

Comment:
Does it make sense/possible to build the debugging into the fanouts themselves?

Reply:
Thanks for the suggestion, it would be ideal if we could just modify the fanouts and get support for all the components using them but the livedebugging service still needs to communicate with the components to notify them when a livedebugging session is opened/closed. So in the end I believe that it wouldn't improve the code much. Also the fanouts are copies from upstream. Modifying copied code makes dependency updates harder and prevents us from easily removing the copied code if it's moved out of an internal pkg from upstream. This is only from the top of my head, it could be worth digging through it to maybe find a simpler way